sos/plugins/vdsm.py

```diff
@@ -0,0 +1,153 @@
+
+
+class Vdsm(Plugin, RedHatPlugin):
+    """vdsm Plugin"""
+
+    packages = (
+        'vdsm',
+        'vdsm-client',
+    )
+
+    plugin_name = 'vdsm'
+
+    optionList = [("logsize", 'max size (MiB) to collect per log file', '',
+                   False)]
+
+    def setup(self):
+        os.environ["LVM_SYSTEM_DIR"] = "/var/run/vdsm/lvm"
+        self.add_forbidden_path('/etc/pki/vdsm/keys/*')
+        self.add_forbidden_path('/etc/pki/vdsm/libvirt-spice/*-key.*')
+        self.add_forbidden_path('/etc/pki/libvirt/private/*')
+
+        self.add_cmd_output('service vdsmd status')
+
+        self.add_copy_spec('/tmp/vds_installer*')
+        self.add_copy_spec('/tmp/vds_bootstrap*')
+        self.add_copy_spec('/etc/vdsm/*')
+
+        logsize = self.get_option('logsize')
+        if logsize is not None:
+            self.add_copy_spec('/var/log/vdsm/*', logsize)
+        else:
+            self.add_copy_spec('/var/log/vdsm/*')
+
+        self._add_vdsm_run_dir()
+        self.add_copy_spec('/usr/libexec/vdsm/hooks')
+        self.add_copy_spec('/var/lib/vdsm')
+        self.add_copy_spec('/var/log/ovirt.log')
+        self.add_copy_spec('/var/log/sanlock.log')
+
+        res = self.call_ext_prog('/usr/bin/pgrep qemu-kvm')
+        if res['status'] == 0:
+            for line in res['output']:
+                pid = line.strip()
+                self.add_copy_spec("/proc/%s/cmdline" % pid)
+                self.add_copy_spec("/proc/%s/status" % pid)
+                self.add_copy_spec("/proc/%s/mountstats" % pid)
+        self.add_cmd_output("/bin/ls -ldZ /etc/vdsm")
+        self.add_cmd_output(
+            "/bin/su vdsm -s /bin/sh -c '/usr/bin/tree -l /rhev/data-center'")
+        self.add_cmd_output(
+            "/bin/su vdsm -s /bin/sh -c '/bin/ls -lR /rhev/data-center'")
+        self.add_cmd_output(
+            "/sbin/lvm vgs -v -o +tags --config \'%s\'" % LVM_CONFIG)
+        self.add_cmd_output(
+            "/sbin/lvm lvs -v -o +tags --config \'%s\'" % LVM_CONFIG)
+        self.add_cmd_output(
+            "/sbin/lvm pvs -v -o +all --config \'%s\'" % LVM_CONFIG)
+
+        self.add_cmd_output("/sbin/fdisk -l")
+        self.add_cmd_output("/usr/bin/iostat")
+        self.add_cmd_output("/sbin/iscsiadm -m node")
+        self.add_cmd_output("/sbin/iscsiadm -m session")
+        self.add_cmd_output("/usr/sbin/nodectl info")
+        self.add_cmd_output("/usr/bin/abrt-cli list")
+
+        self.add_cmd_output('vdsm-client Host getCapabilities')
+        self.add_cmd_output('vdsm-client Host getStats')
+        self.add_cmd_output('vdsm-client Host getAllVmStats')
+        self.add_cmd_output('vdsm-client Host getVMList')
+        self.add_cmd_output('vdsm-client Host getDeviceList')
+        self.add_cmd_output('vdsm-client Host hostdevListByCaps')
+        self.add_cmd_output('vdsm-client Host getAllTasksInfo')
+        self.add_cmd_output('vdsm-client Host getAllTasksStatuses')
```

Comment:
Same here, or we could have the list contain subcmds and iterate through those, e.g.:

```python
subcmds= [
    'getStats',
    [...]
]

for subcmd in subcmds:
    self.add_cmd_output('vdsm-client Host %s' % subcmd)
```